### DISCUSSION TITLE
Mona Lisa Art Auction inspired by tinybid

### DISCUSSION THREAD

**Discussion by daningyn** [NillionNetwork/discussions#67] [Show and tell]
### Show and tell project type

Builder Bounty Submission

### Github Repo Link

https://github.com/daningyn/tinybid-nillion-blind-app

### Video Walkthough Link

https://vimeo.com/966608234

### Project Description

Vickrey Auction: Secure Single-Item First-Price Auction inspired by tiny bid
Utilizing the advanced capabilities of the Nada program and the innovative technology of Nillion, the app guarantees that all bids remain confidential while accurately determining the highest bidder

### What problems does your project solve? How does it preserve privacy for users?

- Problem
In a traditional Vickrey auction, bidders submit sealed bids, and the highest bidder wins. However, there are significant challenges in maintaining the confidentiality of the bids. Ensuring that each bid remains confidential until the auction ends is crucial. Any leak of bid information can lead to unfair advantages and undermine the auction's integrity

- Solution
Thanks to Nillion technology and Nada program, it encrypts each bid, ensuring that only authorized computations can decrypt and process the bids. This encryption is maintained throughout the bidding process, preventing any leaks or unauthorized access to the bid values. Nillion's technology enables secure multi-party computation (MPC), which allows the app to perform calculations on encrypted data without needing to decrypt it.

### How does the project use Nillion? Describe and link to any Nada programs

I used Nada program for protecting bid data and calculations without decryption
Here is my nada program: https://github.com/daningyn/tinybid-nillion-blind-app/blob/master/packages/nillion/next-project-programs/src/nft_bid.py

### Is there anything else you want to share?

In my previous work with the FHE Inco Network project, I utilized Fully Homomorphic Encryption (FHE) to secure the bidding process in Vickrey auctions for NFTs. While FHE provided robust security, I encountered significant challenges with gas costs associated with encryption and decryption, with some transactions reaching up to 2 million gas. This high cost rendered the solution impractical for widespread use. 

You can refer it here: https://inftco.daningyn.xyz/

I hope that I can transform this idea into a reality that not only addresses the current limitations but also sets a new standard for secure and efficient online bidding. I am eager to contribute to this vision to the Nillion's community projects on github if possible. I love to be a part of this data war.

### Optional - Link your project and team members' social handles

https://x.com/d4nhngyn

### Optional - Team ETH Address(es)

- 0x14a5095560250146cc3aadb5a7501d7b49d544c7

**Comment by oceans404** [comment]
Congratulations @daningyn!!! You won our tinybid bounty. Can you confirm that the ETH address you provided is correct so we can send your prize: 0x14a5095560250146cc3aadb5a7501d7b49d544c7

tinydemo winners announcement tweet: https://x.com/nillionnetwork/status/1820606184895947087

**Reply by daningyn** [reply]
Hi @oceans404, thank you!! I confirm that the ETH address `0x14a5095560250146cc3aadb5a7501d7b49d544c7` is correct

**Reply by Nillion-Engineering** [reply]
Hey, Claire from the Nillion Association here! Thanks for building on Nillion. Here's the link to your prize. Well done!

https://etherscan.io/tx/0x45d82d9f38427c40a1705c6251559f6d738b3e0506d5f49515dde6305f5b0033
